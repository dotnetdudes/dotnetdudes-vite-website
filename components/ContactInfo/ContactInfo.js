// <ContactInfo></ContactInfo>
import ContactInfoTemplate from './template';

class ContactInfo extends HTMLElement {
    constructor() {
        super();
    }

    encMail = 'am9obkBkb3RuZXRkdWRlcy5jb20=';

    getEmail() {
        return atob(this.encMail);
    }

    sendEmail() {
        const email = this.getEmail();
        window.location = `mailto:${email}?subject=Dudes%20Contact%20Form`;
    }

    callDudes() {
        window.location = `tel:+61402888801`;
    }

    dudesGithub() {
        window.open('https://github.com/dotnetdudes', '_blank');
    }

    dudesLinkedIn() {
        window.open(
            'https://www.linkedin.com/company/3506154/admin/feed/posts/',
            '_blank'
        );
    }

    connectedCallback() {
        this.innerHTML = ContactInfoTemplate;

        const sendMessage = () => this.sendEmail();
        const phoneDudes = () => this.callDudes();
        const gotoGithub = () => this.dudesGithub();
        const gotoLinkedin = () => this.dudesLinkedIn();

        const email = this.getEmail();
        const emailInfo = document.getElementById('email-item');
        emailInfo.innerText = email;
        // click event on email field
        emailInfo.addEventListener('click', () => {
            sendMessage();
        });

        const mobileInfo = document.getElementById('mobile-item');
        mobileInfo.addEventListener('click', () => {
            phoneDudes();
        });

        const githubInfo = document.getElementById('github-item');
        githubInfo.addEventListener('click', () => {
            gotoGithub();
        });

        const linkedinInfo = document.getElementById('linkedin-item');
        linkedinInfo.addEventListener('click', () => {
            gotoLinkedin();
        });
    }
}

window.customElements.define('contact-info', ContactInfo);
