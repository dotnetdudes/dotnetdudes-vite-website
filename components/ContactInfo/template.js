const ContactInfoTemplate = /*html*/ `
<ul class="contact-info">
    <li class="contact-info-item email-item" id="email-item">Email</li>
    <li class="contact-info-item mobile-item" id="mobile-item">+61 402 888 801</li>
    <li class="contact-info-item github-item" id="github-item">Github</li>
    <li class="contact-info-item linkedin-item" id="linkedin-item">LinkedIn</li>
    <li class="contact-info-item facebook-item" id="facebook-item">facebook</li>
    <li class="contact-info-item twitter-item" id="twitter-item">Twitter</li>
</ul>
`;

export default ContactInfoTemplate;