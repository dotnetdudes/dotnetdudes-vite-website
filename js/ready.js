// import { getInitialTheme, setTheme } from './theme';

export const onSiteLoaded = () => {
    try {

        document.addEventListener('DOMContentLoaded', function () {
            // const theme = getInitialTheme();
            // console.log(theme);
            // setTheme(theme);
            // const pnl = document.getElementById('loadingpanel');
            // pnl.style.display = 'none';
            console.log('DOMContentLoaded');
        });
    } catch (error) {
        console.log(error);
    }
};
