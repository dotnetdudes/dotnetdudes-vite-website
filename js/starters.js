export const onStarterLoaded = () => {
    try {
        const openReactStarter = () => {
            window.open("https://github.com/dotnetdudes/react-starter", "_blank");
        }
        const openReactNativeStarter = () => {
            window.open("https://github.com/dotnetdudes/react-native-starter", "_blank");
        }
        document.addEventListener('DOMContentLoaded', function () {
            const btn = document.getElementById('reactStarter');
            const reactBtn = document.getElementById('nativeStarter');
            btn.addEventListener('click', openReactStarter);
            reactBtn.addEventListener('click', openReactNativeStarter);
        });
    } catch (error) {
        console.log(error);
    }
}
