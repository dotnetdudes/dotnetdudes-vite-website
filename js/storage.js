export const setItem = (name, item) => {
    if (!window.localStorage) return false;
    window.localStorage.setItem(name, item);
    return true;
};

export const getItem = (name) => {
    if (!window.localStorage) return false;
    return window.localStorage.getItem(name);
};

export const clearStorage = () => {
    if (!window.localStorage) return false;
    window.localStorage.clear();
};
