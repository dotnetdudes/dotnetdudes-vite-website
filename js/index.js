import { onSiteLoaded } from './ready';
/* import dotenv from 'dotenv';
const dotenv_result = dotenv.config({ silent: true }); */

// onSiteLoaded();
console.log('okay');
