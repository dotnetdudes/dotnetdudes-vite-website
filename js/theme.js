import { getItem, setItem } from "./storage";

export const isDarkMode = () =>
  window.matchMedia &&
  window.matchMedia("(prefers-color-scheme: dark)").matches;

export const getInitialTheme = () => {
  let initialTheme = isDarkMode() ? "dark" : "light";
  // console.log(isDarkMode());
  if (getItem("dudesTheme")) initialTheme = getItem("dudesTheme");
  return initialTheme;
};

export const setTheme = (theme) => {
  // set css link href
  setItem("dudesTheme", theme);
};
