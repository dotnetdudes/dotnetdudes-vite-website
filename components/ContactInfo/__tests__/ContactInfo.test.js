import { describe, test, expect, beforeEach, afterEach, vi } from 'vitest';
import { screen } from '@testing-library/dom';
import '../ContactInfo.js';

describe('ContactInfo', () => {
  beforeEach(() => {
    vi.stubGlobal('open', vi.fn());
  });

  afterEach(() => {
    vi.clearAllMocks();
    vi.resetAllMocks();
  });

  // snapshot test
  test('should match current snapshot', () => {
    document.body.innerHTML = `<contact-info></contact-info>`;
    expect(document.body.innerHTML).toMatchSnapshot();
  });

  test('should decode and render email', () => {
    document.body.innerHTML = `<contact-info></contact-info>`;
    const email = document.getElementById('email-item');
    expect(email.innerText).toBe('john@dotnetdudes.com');
  });

  test('should open Github link', () => {
    document.body.innerHTML = `<contact-info></contact-info>`;
    const spy = vi.spyOn(window, 'open');
    const github = screen.getByText('Github');

    github.click();

    expect(spy).toHaveBeenCalledWith(
      'https://github.com/dotnetdudes',
      '_blank',
    );
  });

  test('should open LinkedIn link', () => {
    document.body.innerHTML = `<contact-info></contact-info>`;
    const spy = vi.spyOn(window, 'open');
    const linkedin = screen.getByText('LinkedIn');

    linkedin.click();

    expect(spy).toHaveBeenCalledWith(
      'https://www.linkedin.com/company/3506154/admin/feed/posts/',
      '_blank',
    );
  });

  test('should open email link', () => {
    document.body.innerHTML = `<contact-info></contact-info>`;
    const spy = vi.spyOn(window, 'location', 'set');
    const email = screen.getByText('Email');

    email.click();

    expect(spy).toHaveBeenCalledWith(
      'mailto:john@dotnetdudes.com?subject=Dudes%20Contact%20Form',
    );
  });

  test('should open phone link', () => {
    document.body.innerHTML = `<contact-info></contact-info>`;
    const spy = vi.spyOn(window, 'location', 'set');
    const phone = screen.getByText('+61 402 888 801');

    phone.click();

    expect(spy).toHaveBeenCalledWith('tel:+61402888801');
  });
});
