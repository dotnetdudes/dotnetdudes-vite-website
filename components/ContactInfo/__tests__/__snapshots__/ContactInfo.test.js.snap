// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`should match current snapshot 1`] = `
"<contact-info>
<ul class="contact-info">
    <li class="contact-info-item email-item" id="email-item">Email</li>
    <li class="contact-info-item mobile-item" id="mobile-item">+61 402 888 801</li>
    <li class="contact-info-item github-item" id="github-item">Github</li>
    <li class="contact-info-item linkedin-item" id="linkedin-item">LinkedIn</li>
    <li class="contact-info-item facebook-item" id="facebook-item">facebook</li>
    <li class="contact-info-item twitter-item" id="twitter-item">Twitter</li>
</ul>
</contact-info>"
`;
